### ISSUE TITLE
[DOC] - Add a11y documentation to the JupyterLab's contributor docs

### ISSUE THREAD

**Issue by trallard** [Quansight-Labs/jupyter-a11y-mgmt#18]
## Background

Improve Jupyterlab's contributor documentation to account for accessibility testing and best practices as part of the workflow.

## Expected timeline

We can also document as we do work on our Team Compass
Should be done by the 18 months milestone

```[tasklist]
### Identified content
- [x] WCAG intro and links to resources - @isabela-pf @steff456
- [x] What to look for when reviewing PRs 📌  - design @isabela-pf (mentioned on [jupyter/accessibility #125 comment](https://github.com/jupyter/accessibility/issues/125#issuecomment-1511706493))
- [x] Write up results from #146 and #147 in [iota-school/notebooks-for-all](https://github.com/Iota-School/notebooks-for-all). Link to user testing in [jupyter/accessibility](https://github.com/jupyter/accessibility/) and [jupyter/surveys](https://github.com/jupyter/surveys)
- [x] Write up results from #184 in [quansight-labs/jupyterlab-user-testing](https://github.com/quansight-Labs/jupyterlab-user-testing/issues) and/or [jupyter/accessibility](https://github.com/jupyter/accessibility/). Link to user testing in [jupyter/accessibility](https://github.com/jupyter/accessibility/) and [jupyter/surveys](https://github.com/jupyter/surveys)
```
```[tasklist]
### Future work
- [ ] Later [based on our automated actions](https://github.com/gabalafou/jupyterlab/actions/runs/3410796815) - how to use the automated tests
- [ ] Best practices -> use colour variables - https://github.com/jupyterlab/jupyterlab/issues/14399
```
